soco/music_library.py

```diff
@@ -288,14 +288,20 @@ def get_music_library_information(
              `SoCoException` upon errors.
         """
         search = self.SEARCH_TRANSLATION[search_type]
+        share_translation = self.SEARCH_TRANSLATION["share"]
 
         # Add sub categories
-        if subcategories is not None:
+        # sub categories are not allowed when searching shares
+        if subcategories is not None and search_term != share_translation:
```

Comment:
I **think** something is wrong here. `search_term` is a word or network path or similar, so I don't think that should be compared to `share_translation`. I think the intention must have been (I'm aware that you are implementing it from the suggestions in the issue) it should have read `search != share_translation`.

But actually, I would prefer doing these tests differently. It's a little bit the long way around to introduce a new variable and compare to the search, when really what determines the changes in behavior is the `search_type`. So I would prefer it if you would remove the extra variable and rewrite the test here and below against `search_type`.

Reply:
I also think you're right with suggesting
`if subcategories is not None and search != share_translation:`
My original proposal
`if subcategories is not None and search_term != share_translation:`
was wrong
FWIW I introduced the share_translation variable to make it readable (for me!).
It could easily be
`if subcategories is not None and search != "S:":`
or, if you want to use search_type
`if subcategories is not None and search_type != "share":`
From your comment I presume you favour the last option
I'm happy with that

Reply:
Modded as per comments.

Reply:
@Sonosy great that we got that cleared up. Yes, I favor the latter, but that is because it is already established behavior. The `search_type` exactly describes which search type we are doing, which is what you are looking for, and it's option is described in the doc string. In any case, thanks a lot for the information and the fix. We will get it merged in no time.